nixos/modules/system/boot/luksroot.nix

```diff
@@ -470,6 +571,37 @@ in
             '';
           };
 
+          gpgCard = mkOption {
+            default = null;
+            description = ''
+              The option to use this LUKS device with a GPG encrypted luks password by the GPG Smartcard.
+              If null (the default), GPG-Smartcard will be disabled for this device.
+              Yubikey and GPG Card may not be used at the same time.
```

Comment:
If you can't use yubikey and gpg card at the same time, you may want to add an assertion to `config.assertions`.

Reply:
@artemist I have added the assertion.